### ISSUE TITLE
[DOC] Reference missing in AEAttentionBiGRUClusterer

### ISSUE THREAD

**Issue by PipaFlores** [aeon-toolkit/aeon#3021]
### Describe the issue linked to the documentation

Hello!

I was interested in using the implementation of the [AEAttentionBiGRUClusterer](https://www.aeon-toolkit.org/en/stable/api_reference/auto_generated/aeon.networks.AEAttentionBiGRUNetwork.html#aeon.networks.AEAttentionBiGRUNetwork), which references to the DeTSEC paper of  Ienco, D., & Interdonato, R. (2020). However, the implementation in this library does not include the cluster optimization part, which is core of most of the Deep clustering papers theme. (DEC, and DeTSEC).

I am more versed in pytorch than keras, but by thoroughly inspecting the code is easy to see that the clustering is just fitting to data, not optimizing the manifold to it. 

### Suggest a potential alternative/fix

I suggest that this is written explicitly in the documentation. For example, describe the clustering procedure as an GRU-Attention auto-encoder RNN + chosen clustering estimator (by default TimeSeriesKMeans). Or as the first phase of the referenced paper. But without any details in docs,  and a review, users might think that they are implementing the full pipeline described in the paper

I would love to try and make the full implementation, but since pytorch is not the standard here, I doubt that I will invest the time in learning Keras.

**Comment by TonyBagnall** [comment]
thanks for pointing this out, I'm not familiar with this algorithm, any thoughts @hadifawaz1999 ?

**Comment by hadifawaz1999** [comment]
Hello, thanks for raising the issue. In fact it is not a bug or wrong implementation. This is simply a missing reference issue.
The reference that should be added is:

Lafabregue, Baptiste, Jonathan Weber, Pierre Gançarski, and Germain Forestier. "End-to-end deep representation learning for time series clustering: a comparative study." Data mining and knowledge discovery 36, no. 1 (2022): 29-81.

As they proposed a benchmark comparing deep learning clustering models, including architecture, clustering losses and pre-text losses.
They found that generally, using any clustering loss doesnt really help much and that the best pre-text losses are reconstruction and multi-reconstruction losses.
However they also took the network from Ienco, D., & Interdonato, R.'s paper "Deep Multivariate Time Series Embedding Clustering via Attentive-Gated Autoencoder" and used it with different clustering loss and pre-text loss. So we included the [AEAttentionBiGRUClusterer](https://github.com/aeon-toolkit/aeon/blob/main/aeon/clustering/deep_learning/_ae_abgru.py) version with the winning strategy thats all. If you want to PR the addional reference and mention this in the docstring you're welcome to that would be great.
